src/handle_action.cpp

```diff
@@ -1318,14 +1318,18 @@ static void fire()
         std::vector<std::function<void()>> actions;
 
         for( auto &w : u.worn ) {
-            if( w.type->can_use( "holster" ) && !w.has_flag( flag_NO_QUICKDRAW ) &&
-                !w.contents.empty() && w.contents.front().is_gun() ) {
+
+            std::vector<item *> guns = w.items_with( []( const item & it ) {
```

Comment:
note to self, this visits itself first. will this be an issue with for instanced strapped guns?